### PR TITLE
feat: [2U-296] add help tokens

### PR DESCRIPTION
**Pull request by Thposbsmt**
Ticket: https://youtrack.raccoongang.com/issue/2U-296

### CODE REVIEW

File: src/course-outline/outline-sidebar/OutlineSidebar.jsx [hunk 1]
```diff
@@ -2,19 +2,20 @@ import React from 'react';
...
 import HelpSidebar from '../../generic/help-sidebar';
-import { getOutlineIndexData } from '../data/selectors';
 import { getFormattedSidebarMessages } from './utils';
+import { useHelpTokens } from '../../help-tokens/hooks';
 
 const OutlineSideBar = ({ courseId }) => {
   const intl = useIntl();
+
   const {
-    learnMoreGradingUrl,
```
Comment (ruzniaievdm) [inline src/course-outline/outline-sidebar/OutlineSidebar.jsx:14]: please remove these here too `frontend-app-course-authoring/src/course-outline/__mocks__/courseOutlineIndex.js`

**Review by ruzniaievdm — CHANGES_REQUESTED** [review]

File: src/help-tokens/hooks.test.jsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import { useDispatch, useSelector } from 'react-redux';
+import { renderHook } from '@testing-library/react-hooks';
+import { useHelpTokens } from './hooks';
+import { RequestStatus } from '../data/constants';
+import { fetchHelpTokens } from './data/thunks';
```
Comment (PKulkoRaccoonGang) [inline src/help-tokens/hooks.test.jsx:5]:
```suggestion
import { useDispatch, useSelector } from 'react-redux';
import { renderHook } from '@testing-library/react-hooks';

import { RequestStatus } from '../data/constants';
import { useHelpTokens } from './hooks';
import { fetchHelpTokens } from './data/thunks';
```

File: src/help-urls/data/slice.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
+/* eslint-disable no-param-reassign */
+import { createSlice } from '@reduxjs/toolkit';
+
+import { RequestStatus } from '../../data/constants';
+
+const initialState = {
+  loadingHelpUrlsStatus: RequestStatus.PENDING,
```
Comment (ruzniaievdm) [inline src/help-urls/data/slice.js:7]: let's use `RequestStatus.IN_PROGRESS` for loading initial value

Comment (Thposbsmt) [reply src/help-urls/data/slice.js:7]: Why? I don't understand why everyone use `IN_PROGRESS` when logically it's `PENDING`, hadn't started

Comment (Thposbsmt) [reply src/help-urls/data/slice.js:7]: How I can handle if request was started if request in progress in initial state and in progress when really in progess?

Comment (Thposbsmt) [reply src/help-urls/data/slice.js:7]:
```
Pending: Work on the objective has not yet started.
In Progress: Work has started toward objective completion
```

Comment (ruzniaievdm) [reply src/help-urls/data/slice.js:7]: Actually I agree with you it looks a little confusing)
`frontend-app-course-authoring/src/custom-pages/data/thunks.js`

they use status:
- `RequestStatus.IN_PROGRESS` for loading using `GET`
- `RequestStatus.PENDING` for saving using `POST`, `DELETE`, ...

in this case `slice.js` looks like this
```
loadingStatus: RequestStatus.IN_PROGRESS,
savingStatus: '',
```

Comment (Thposbsmt) [reply src/help-urls/data/slice.js:7]: Ok, I'll change it to `IN_PROGRESS`

Comment (ruzniaievdm) [reply src/help-urls/data/slice.js:7]: Alternatively, we can use `''` for both initial
`IN_PROGRESS` for starting loading query 
`PENDING ` for starting saving query

**Review by ruzniaievdm — APPROVED** [review]

File: src/grading-settings/grading-sidebar/GradingSidebar.test.jsx [hunk 4]
```diff
@@ -1,10 +1,13 @@
 import React from 'react';
 import { IntlProvider, injectIntl } from '@edx/frontend-platform/i18n';
 import { render } from '@testing-library/react';
-
+import { AppProvider } from '@edx/frontend-platform/react';
+import { initializeMockApp } from '@edx/frontend-platform';
+import initializeStore from '../../store';
```
Comment (PKulkoRaccoonGang) [inline src/grading-settings/grading-sidebar/GradingSidebar.test.jsx:6]: Please, add an empty line between library imports and local imports

```suggestion
import { initializeMockApp } from '@edx/frontend-platform';

import initializeStore from '../../store';
```

**Review by PKulkoRaccoonGang — APPROVED** [review]

**Comment by PKulkoRaccoonGang** [comment]
@Thposbsmt 1 comment please

**Review by wowkalucky — APPROVED** [review]
OK.

**Pull request merged**